Utilities/precommit.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
+import sys
+import subprocess 
+import datetime
+import argparse
+import commands
+
+
+git_repo_path = os.path.abspath(subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).strip())
+
+
+def splitall(path):
+    allparts = []
+    while 1:
+        parts = os.path.split(path)
+        if parts[0] == path:  # sentinel for absolute paths
+            allparts.insert(0, parts[0])
+            break
+        elif parts[1] == path: # sentinel for relative paths
+            allparts.insert(0, parts[1])
+            break
+        else:
+            path = parts[0]
+            allparts.insert(0, parts[1])
+    return allparts
+    
+    
+def process(filepath):
+    with open(filepath, 'rt') as f:
+        text = f.read()
+    original_text = text
+    
+    lines = text.split('\n')
+    while lines and lines[0].startswith('//'):
+        lines = lines[1:]
+    text = '\n'.join(lines)
+    text = text.strip()
+
+    header = '''//
+//  %s
+//
+//  Copyright (c) %s Open Whisper Systems. All rights reserved.
+//
+
+''' % ( 
+    os.path.basename(filepath),
+    datetime.datetime.now().year,
+    )
+    text = header + text + '\n'
+
+    if original_text == text:
+        return
+    
+    short_filepath = filepath[len(git_repo_path):]
+    if short_filepath.startswith(os.sep):
+       short_filepath = short_filepath[len(os.sep):] 
+    print 'Updating:', short_filepath
+    
+    with open(filepath, 'wt') as f:
+        f.write(text)
+
+
+def should_ignore_path(path):
+    ignore_paths = [
+        os.path.join(git_repo_path, '.git')
+    ]
+    for ignore_path in ignore_paths:
+        if path.startswith(ignore_path):
+            return True
+    for component in splitall(path):
+        if component.startswith('.'):
+            return True
+        if component.endswith('.framework'):
+            return True
+        if component in ('Pods', 'ThirdParty',):
```

Comment:
Assuming we want to use this script wholesale in Signal-iOS, makes sense to add `Carthage`